bin/spack.bat

```diff
@@ -141,21 +144,22 @@ goto :end_switch
 
 :case_env
 :: If no args or args contain --bat or -h/--help: just execute.
+set args_no_quote=%_sp_args:"=%
 if NOT defined _sp_args (
     goto :default_case
-)else if NOT "%_sp_args%"=="%_sp_args:--help=%" (
+)else if NOT "%args_no_quote%"=="%args_no_quote:--help=%" (
```

Comment:
I'm having trouble seeing why `args_no_quote` is needed for this particular check: couldn't `--help` be removed from `_sp_args` directly and compare that with `_sp_args`?

Reply:
See my previous comment.

Reply:
In this case the comparison would be "weird" but not strictly incorrect even if the quotes were kept, right? In that case you'd be comparing

```
""... --help"" == ""...""
```

or if there was no `--help` option

```
""..."" == ""...""
```

(i.e. the statement would be true when we expect and false when we expect)

Reply:
You are right, this part works with both args_no_quote and with _sp_args. The part that only works with args_no_quote is 
`python %spack% %_sp_flags% env activate --bat %args_no_quote:activate=%`
With _sp_args this becomes:
`python C:\Users\dan.lipsa\projects\spack\bin\spack env activate --bat "" "c:\Users\dan.lipsa\projects\spack\cache-env"`
which results in an error:
`Error: unrecognized arguments: c:\Users\dan.lipsa\projects\spack\cache-env`

Reply:
We could get rid of args_no_quote everywhere else if you prefer.

Reply:
Another way to fix this is to use
`python %spack% %_sp_flags% env activate --bat %_sp_args:"activate"=%`
Note I used _sp_args but I remove "activate" including quotes.

Reply:
I think all the if checks should be consistent regarding testing the value of `_sp_args` or `args_no_quote`: they can all do it one way or all do it the other.

Reply:
Yes, this is why I used args_no_quote everywhere. So I'll leave it as is then.